src/Exception.php

```diff
@@ -62,6 +62,15 @@ public function __construct(
             // message contain additional parameters
             $this->params = $message;
             $message = array_shift($this->params);
+            if ($this->params['solutions'] ?? null) {
+                if (!is_array($this->params['solutions'])) {
+                    $this->params['solutions'] = [$this->params['solutions']];
+                }
+                foreach ($this->params['solutions'] as $solution) {
```

Comment:
`foreach ((array)$this->params['solutions'] as $solution)` should be enought, no need for the first `if`.